### PR TITLE
Minor fixes to bindToSocket

### PR DESCRIPTION
**Pull request by jasta**
Make bindToSocket static, mark it non-null and instead throw on error,
and clarify the "retry" semantic to actually indicate the number of
retries not number of tries.

### CODE REVIEW

File: stetho/src/main/java/com/facebook/stetho/server/LocalSocketHttpServer.java [hunk 1]
```diff
@@ -266,27 +264,36 @@ public void stop() {
...
+      }
+    } while (retries-- > 0);
+
+    throw firstException;
+  }
+
+  private static void sleepUninterruptibly(long millis) {
+    while (true) {
+      try {
+        Thread.sleep(millis);
```
Comment (longinoa) [inline stetho/src/main/java/com/facebook/stetho/server/LocalSocketHttpServer.java]: You should save state here. If you keep getting interrupted you will never terminate this loop.

File: stetho/src/main/java/com/facebook/stetho/server/LocalSocketHttpServer.java [hunk 2]
```diff
@@ -266,27 +264,36 @@ public void stop() {
...
+    int retries = MAX_BIND_RETRIES;
+    IOException firstException = null;
+    do {
+      try {
+        if (LogUtil.isLoggable(Log.DEBUG)) {
+          LogUtil.d("Trying to bind to @" + address);
+        }
+        return new LocalServerSocket(address);
+      } catch (BindException be) {
+        LogUtil.w(be, "Binding error, sleep 1 second ...");
```
Comment (longinoa) [inline stetho/src/main/java/com/facebook/stetho/server/LocalSocketHttpServer.java]: prefer to use the constant `TIME_BETWEEN_RETRIES_MS`

**Comment by longinoa** [comment]
Can you add a description as well.

File: stetho/src/main/java/com/facebook/stetho/server/LocalSocketHttpServer.java (hunk 1 above)
Comment (jasta) [inline stetho/src/main/java/com/facebook/stetho/server/LocalSocketHttpServer.java]: Ack'd, wasn't thinking...

**Comment by jasta** [comment]
Updated.  :+1: or :-1: ?

**Comment by longinoa** [comment]
:-1: you still sleep for `millis` rather than `remaining`

**Comment by longinoa** [comment]
:+1:

**Pull request merged**